app/src/test/java/ingsw/group1/findmyphone/location/GeoPositionTest.java

```diff
@@ -0,0 +1,176 @@
+        assertEquals(exampleLatitude, examplePosition.getLatitude());
+    }
+
+    /**
+     * Testing {@link GeoPosition#getLongitude()} returns the actual value of the longitude.
+     */
+    @Test
+    public void getLongitude_returnsActual() {
+        assertEquals(exampleLongitude, examplePosition.getLongitude());
+    }
+
+    /**
+     * Testing {@link GeoPosition#setLatitude(double)} sets a new value of the latitude.
+     */
+    @Test
+    public void setLatitude_setsCorrectly() {
+        double expectedLatitude = exampleLatitude * 2;
+        examplePosition.setLatitude(expectedLatitude);
+        assertEquals(expectedLatitude, examplePosition.getLatitude());
+    }
+
+    /**
+     * Testing {@link GeoPosition#setLongitude(double)} sets a new value of the longitude.
+     */
+    @Test
+    public void setLongitude_setsCorrectly() {
+        double expectedLongitude = exampleLongitude * 2;
+        examplePosition.setLongitude(expectedLongitude);
+        assertEquals(expectedLongitude, examplePosition.getLongitude());
+    }
+
+    /**
+     * Testing {@link GeoPosition#equals(Object)} returns false for a non {@link GeoPosition}
+     * Object.
+     */
+    @Test
+    public void equals_falseForNonPosition() {
+        assertNotEquals(examplePosition, "Hello there");
+    }
+
+    /**
+     * Testing {@link GeoPosition#equals(Object)} returns false for a {@link GeoPosition} Object
+     * that does not match the equation criteria.
+     */
+    @Test
+    public void equals_falseForNonEqualPosition() {
+        //Latitude and longitude are switched.
+        GeoPosition notTheSamePosition = new GeoPosition(
+                exampleLongitude,
+                exampleLatitude
+        );
+        assertNotEquals(notTheSamePosition, examplePosition);
+    }
+
+    /**
+     * Testing {@link GeoPosition#equals(Object)} returns true for a {@link GeoPosition} Object
+     * that does match the equation criteria.
+     */
+    @Test
+    public void equals_trueForEqualPosition() {
+        GeoPosition theSamePosition = new GeoPosition(
+                exampleLatitude,
+                exampleLongitude
+        );
+        assertEquals(examplePosition, theSamePosition);
+    }
+
+    /**
+     * Testing that {@link GeoPosition#toString()} returns a {@link String} formatted as stated
+     * in the specs:
+     * [latitude]{@link GeoPosition#POSITION_SPLIT_SEQUENCE}[longitude]
+     */
+    @Test
+    public void toString_actsAsStated() {
+        String splitSequence = GeoPosition.POSITION_SPLIT_SEQUENCE;
+        String expectedString = exampleLatitude + splitSequence + exampleLongitude;
+        assertEquals(expectedString, examplePosition.toString());
+    }
+
+    /**
+     * Testing that {@link GeoPosition#GeoPosition(String)} does not accept a String that splits
+     * into more than two parts.
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void stringConstructor_failsForInvalidSplitString() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'stringConstructor_failsForInvalidSplitString' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4950325)